### PR TITLE
Mpuncel/uuid hooks

### PR DESCRIPTION
**Pull request by mpuncel**
Note to reviewers, this has commits from #675 and #677 so review those first

### CODE REVIEW

**Comment by rudle** [comment]
This has compilation errors in travis

**Comment by mpuncel** [comment]
whoops

**Comment by anthonybishopric** [comment]
🚢

File: pkg/preparer/orchestrate_test.go [hunk 1]
```diff
@@ -89,6 +89,10 @@ func (t *TestPod) Home() string {
 	return os.TempDir()
 }
 
+func (t *TestPod) UniqueKey() types.PodUniqueKey {
+	return types.PodUniqueKey(t.currentManifest.ID())
```
Comment (mpuncel) [inline pkg/preparer/orchestrate_test.go]: i wish i returned `""` here instead, changing

**Comment by mpuncel** [comment]
merging on green build

**Pull request merged**